pkg/smokescreen/smokescreen.go

```diff
@@ -455,6 +470,13 @@ func BuildProxy(config *Config) *goproxy.ProxyHttpServer {
 
 	// Handle traditional HTTP proxy
 	proxy.OnRequest().DoFunc(func(req *http.Request, pctx *goproxy.ProxyCtx) (*http.Request, *http.Response) {
+		// Set this on every request as every request mints a new goproxy.ProxyCtx
+		pctx.RoundTripper = rtFn
+
+		// For MITM requests intended for the remote host, the sole requirement was to configure the RoundTripper
```

Comment:
Could you update the comment to include why it doesn't need the rest of this function? I'm not sure I follow.